### PR TITLE
support bigqueryconnection cloudsql connection

### PR DESCRIPTION
**Pull request by 600lyy**
### Change description
Add support for bigqueryconnection cloudsql connection

Run E2E test again realGCP:

```
KCC_USE_DIRECT_RECONCILERS=BigQueryConnectionConnection hack/record-gcp fixtures/cloudsqlconnectionbasic
```

- [] Run `make ready-pr` to ensure this PR is ready for review.
- [x] Perform necessary E2E testing for changed resources.

### CODE REVIEW

File: apis/refs/v1beta1/sqlinstanceref.go [hunk 1]
```diff
@@ -22,3 +34,92 @@ type SQLInstanceRef struct {
 	/* The `namespace` field of a `SQLInstance` resource. */
 	Namespace string `json:"namespace,omitempty"`
 }
+
+type SQLInstance struct {
+	ProjectID       string
+	Location        string
+	SQLInstanceName string
+	ConnectionName  string
```
Comment (yuwenma) [inline apis/refs/v1beta1/sqlinstanceref.go:42]: Normally the Reference struct only holds the fundamental identifies (parent + instance ID). 

Does connection always in the form of `${ProjectID}:${Location}:${SQLInstanceName}`? if so, we can build the ConnectionName from the other three and don't need this field.

File: pkg/controller/direct/bigqueryconnection/connection_mapping.go [hunk 2]
```diff
@@ -36,15 +36,49 @@ func CloudResourcePropertiesSpec_FromProto(mapCtx *direct.MapContext, in *pb.Clo
...
+	}
+	return out
+}
+
+func CloudSqlPropertiesSpec_FromProto(mapCtx *direct.MapContext, in *pb.CloudSqlProperties) *krm.CloudSqlPropertiesSpec {
+	if in == nil {
+		return nil
+	}
+	out := &krm.CloudSqlPropertiesSpec{}
+	// out.InstanceID = direct.LazyPtr(in.InstanceId)
```
Comment (yuwenma) [inline pkg/controller/direct/bigqueryconnection/connection_mapping.go:62]: We want this. You can use assign the value to External 
```
out.InstanceRef = &SQLInstanceRef{
   External: ...
}
```

**Review by yuwenma — COMMENTED** [review]
/approve

Great job! I left some minor comments, and it will be ready to go.

File: apis/refs/v1beta1/sqlinstanceref.go (hunk 1 above)
Comment (600lyy) [reply apis/refs/v1beta1/sqlinstanceref.go:42]: Yeah, makes sense. Even though the form would change in future, I think it still can be built based on other fields.

**Review by yuwenma — COMMENTED** [review]
/lgtm

Thanks! All looks good

**Pull request merged**